### DISCUSSION TITLE
Auto-linking using userIdentity Update

### DISCUSSION THREAD

**Discussion by stephanieakpakoun** [azure-ad-b2c/samples#486] [Q&A]
The [docs on social claims transformations](https://learn.microsoft.com/en-us/azure/active-directory-b2c/social-transformations) and the starter social and local accounts sample user AlternativeSecurityIds whereas the account linking samples specifically the [auto-account-linking sample](https://github.com/azure-ad-b2c/samples/tree/master/policies/auto-account-linking/policy) use UserIdentity to store the social accounts. Which one should we use? It is confusing the see mentions of both. Is there a reason UserIdentity was user instead of alternativeUserIds? If not, could the samples be updated to reflect the current state of the documentation?

**Comment by weyto** [comment]
Hi @stephanieakpakoun 

Does one of the link helps you?
[https://github.com/MicrosoftDocs/azure-docs/issues/32185](url)
[https://github.com/Azure-Samples/active-directory-b2c-advanced-policies/issues/50](url)

My understanding is:
Use `AlternativeSecurityId` instead of `UserIdentity` if you implement new stuff, because it is the newer one, .
